android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java

```diff
@@ -116,12 +117,21 @@ public Bundle getSkuDetails(final int apiVersion, final String packageName, fina
         return response;
       }
     } else {
+      DeveloperPayload developerPayloadObject =
+          new DeveloperPayload(developerPayload, PayloadHelper.getPayload(developerPayload),
+              PayloadHelper.getOrderReference(developerPayload),
+              PayloadHelper.getOrigin(developerPayload));
       final BuyItemProperties buyItemProperties =
-          new BuyItemProperties(apiVersion, packageName, sku, type, developerPayload);
+          new BuyItemProperties(apiVersion, packageName, sku, type, developerPayloadObject);
 
       final Context context = WalletUtils.getContext();
-
-      Intent intent = new Intent(context, IabActivity.class);
+      Intent intent;
+      if (type.equalsIgnoreCase("inapp") && sku != null && !sku.isEmpty()) {
+        intent = new Intent(context, IabActivity.class);
```

Comment:
IabActivity and InstallDialogActivity should have methods to create the intent to open them